src/app/features/location/geo.service.ts

```diff
@@ -69,14 +69,28 @@ export class GeoService {
   private reformatDisplayName(result: OpenStreetMapsSearchResult): GeoResult {
     const addr = result?.address;
     if (addr) {
-      const city = addr.city ?? addr.town;
+      const city = addr.city ?? addr.town ?? "";
+      let street = addr.road ? addr.road : "";
+      if (street && addr.house_number) {
+        street = `${street} ${addr.house_number}`;
+      } else if (addr.house_number) {
+        street = addr.house_number;
+      }
+      let postcodeCity = "";
+      if (addr.postcode && city) {
+        postcodeCity = `${addr.postcode} ${city}`;
+      } else if (addr.postcode) {
+        postcodeCity = `${addr.postcode}`;
+      } else if (city) {
+        postcodeCity = city;
+      }
 
       result.display_name = [
         addr.amenity ?? addr.office,
-        addr.road ? addr.road + " " + addr.house_number : undefined,
-        addr.postcode ? addr.postcode + " " + city : city,
+        street || undefined,
+        postcodeCity || undefined,
       ]
-        .filter((x) => !!x)
+        .filter((x) => !!x && x !== "undefined")
```

Comment:
nice solution overall!

Could you write a unit test in geo.service.spec.ts also? I would suggest we call `service.reformatDisplayName(testResult)` for different values of a testResult variable and check if the resulting string is nice and as expected. The reformatDisplayName is currently private, so the test file will not find it. But you can remove the "private" here from the method also, because this method could be useful even outside of the class.